### PR TITLE
HL-321 | Integrate YRTTI API

### PR DESCRIPTION
**Pull request by quyenlq**
## Description :sparkles:
YRTTI API Integration to query association info via Palveluväylä
Documentation of YRTTI API can be found: https://helsinkisolutionoffice.atlassian.net/wiki/spaces/KAN/pages/7562068053/Working+document+Integration+to+Palveluv+yl+from+app+developer+s+perspective

- Added a YRTTIClient to `shared` component, although AFAIK Kesaseteli does not support application for association.
- Update company query logic so it'll try to get association info if the first query to company return nothing.
- Update unittests

### CODE REVIEW

File: backend/shared/shared/yrtti/yrtti_client.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    def _get_active_association_name(self, name_info: dict) -> str:
+        # There will be the case where an association has more than one name
+        # Here we only pick up latest name available
+        return [
+            name
+            for name in name_info
+            if name["AssociationNameType"] == TARGET_ASSOCIATION_NAME_TYPE
+            and name["AssociationNameLanguage"] == TARGET_ASSOCIATION_NAME_LANGUAGE
+            and name["AssociationNameStatus"] == TARGET_ASSOCIATION_NAME_STATUS
+        ][0]
```
Comment (haavikko) [inline backend/shared/shared/yrtti/yrtti_client.py:69]: Is there always at least one name? Raise meaningful error if no name in data?

File: backend/shared/shared/yrtti/yrtti_client.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    def get_association_data_from_yrtti_data(self, yrtti_data: dict) -> dict:
+        """
+        Get the required company fields from YRTTI data.
+        """
+        association_name = self._get_active_association_name(
+            yrtti_data["AssociationNameInfo"]
+        )
+        # There might be a list of addresses, but it is impossible to identify which one is the primary address from
+        # the data so we just select the first one. Most of the time there is only 1 address
```
Comment (haavikko) [inline backend/shared/shared/yrtti/yrtti_client.py:29]: This solution is OK with me, good to have the comment as documentation

**Review by haavikko — APPROVED** [review]
Please check the comments and update code if you think it needs to, merge at will!

At the testing stage, we should test with more associations if at all possible

File: backend/benefit/helsinkibenefit/urls.py [hunk 3]
```diff
@@ -59,9 +59,6 @@
     path("v1/", include(handler_app_router.urls)),
     path("v1/terms/approve_terms_of_service/", ApproveTermsOfServiceView.as_view()),
     path("v1/company/", GetCompanyView.as_view()),
-    path(
-        "v1/company/<str:business_id>", GetCompanyView.as_view()
-    ),  # FIXME: Remove this later
```
[the remark this answers is not in the document]
Comment (quyenlq) [reply backend/benefit/helsinkibenefit/urls.py:64]: FE guys confirmed that this is not needed anymore

File: backend/shared/shared/yrtti/yrtti_client.py (hunk 1 above)
Comment (quyenlq) [reply backend/shared/shared/yrtti/yrtti_client.py]: Nice catch, from the API doc it is nullable but I don't think it'll ever happen. Updated this so there will be an error if name not found

**Pull request merged**